_posts/pl/2025-01-06-czy-wiesz-czym-jest-partycjonowanie.md

```diff
@@ -0,0 +1,188 @@
+    FOR VALUES IN ('Odzież');
+
+CREATE TABLE furniture PARTITION OF products
+    FOR VALUES IN ('Meble');
+
+-- przykładowe dane
+INSERT INTO products (category, product_name, price)
+VALUES ('Elektronika', 'Telefon', 500.00);
+
+INSERT INTO products (category, product_name, price)
+VALUES ('Odzież', 'T-Shirt', 25.00);
+
+INSERT INTO products (category, product_name, price)
+VALUES ('Meble', 'Sofa', 800.00);
+
+-- Przykładowe zapytania wykorzystujące utworzone partycje
+
+-- Wypisz produkty z kategorii Elektronika
+SELECT * FROM products WHERE category = 'Elektronika';
+
+-- Wypisz produkty z kategorii Odzież
+SELECT * FROM products WHERE category = 'Odzież';
+
+-- Wypisz produkty z kategorii Meble
+SELECT * FROM products WHERE category = 'Meble';
+```
+
+### Partycjonowanie hashowe (ang. Hash partitioning)
+
+Ta metoda partycjonowania polega na podziale danych na partycje na podstawie wartości hash'owej. Każda partycja przechowuje podzbiór danych, które pasują do określonej wartości wspomnianego hasha. Przykładowo, jeśli mamy tabelę zamówień z różnymi identyfikatorami klientów, możemy podzielić tę tabelę na partycje tak, aby każda z nich przechowywała zamówienia o określonej wartości hasha identyfikatora klienta. W ten sposób, chcąc pobrać zamówienia danego klienta, możemy przeszukać tylko jeden, konkretny podzbiór danych.
+
+Przykład:
+```sql
+-- struktura tabeli
+CREATE TABLE orders (
+                        order_id SERIAL PRIMARY KEY,
+                        order_date DATE,
+                        customer_id INT,
+                        total_amount NUMERIC
+) partition by hash(customer_id);
+
+-- tworzenie partycji
+CREATE TABLE orders_1 PARTITION OF orders
+    FOR VALUES WITH (MODULUS 3, REMAINDER 0);
+
+CREATE TABLE orders_2 PARTITION OF orders
+    FOR VALUES WITH (MODULUS 3, REMAINDER 1);
+
+CREATE TABLE orders_3 PARTITION OF orders
+    FOR VALUES WITH (MODULUS 3, REMAINDER 2);
+
+-- przykładowe dane
+INSERT INTO orders (order_date, customer_id, total_amount)
+VALUES ('2023-01-15', 101, 500.00);
+
+INSERT INTO orders (order_date, customer_id, total_amount)
+VALUES ('2023-02-20', 102, 600.00);
+
+INSERT INTO orders (order_date, customer_id, total_amount)
+VALUES ('2023-03-10', 103, 700.00);
+
+-- Przykładowe zapytania wykorzystujące utworzone partycje
+
+-- Pobierz zamówienia dla klienta z ID = 101
+SELECT * FROM orders WHERE customer_id = 101;
+
+-- Pobierz zamówienia dla klienta z ID = 102
+SELECT * FROM orders WHERE customer_id = 102;
+
+-- Pobierz zamówienia dla klienta z ID = 103
+SELECT * FROM orders WHERE customer_id = 103;
+```
+
+## Podsumowanie
+
+Warto zauważyć, że partycjonowanie nie zawsze jest najlepszym rozwiązaniem dla każdego przypadku użycia i ważne jest, aby przeanalizować kompromisy i zdecydować, czy jest to właściwe podejście w konkretnym scenariuszu. Ponadto ważne jest dokładne zaplanowanie i przetestowanie strategii partycjonowania przed wdrożeniem jej w środowisku produkcyjnym.
```

Comment:
Warto zauważyć, że partycjonowanie nie zawsze musi być najlepszym rozwiązanie. Warto pamiętać o tym, aby przeanalizować wszystkie potencjalne wady i zalety wdrożenia tego rozwiązania i na tej podstawie zdecydować, czy jest to właściwe podejście w konkretnym scenariuszu. Ponadto, ważne jest także dokładne zaplanowanie i przetestowanie strategii partycjonowania przed wdrożeniem jej w środowisku produkcyjnym.

Reply:
teraz zobaczyłem, że zostawiłem 2x "warto"
> "Należy pamiętać o tym, aby przeanalizować wszystkie (...)"